build_projects/dotnet-host-build/TestTargets.cs

```diff
@@ -0,0 +1,135 @@
+                .Execute()
+                .EnsureSuccessful();
+
+            return c.Success();
+        }
+
+        [Target]
+        public static BuildTargetResult RestoreTests(BuildTargetContext c)
+        {
+            var dotnet = DotNetCli.Stage0;
+            CleanBinObj(c, Path.Combine(Dirs.RepoRoot, "test"));
+
+            dotnet.Restore("--verbosity", "verbose")
+                .WorkingDirectory(Path.Combine(Dirs.RepoRoot, "test"))
+                .Execute()
+                .EnsureSuccessful();
+            return c.Success();
+        }
+
+        [Target]
+        public static BuildTargetResult BuildTests(BuildTargetContext c)
+        {
+            var dotnet = DotNetCli.Stage0;
+
+            var configuration = c.BuildContext.Get<string>("Configuration");
+
+            foreach (var testProject in TestProjects)
+            {
+                c.Info($"Building tests: {testProject}");
+                dotnet.Build("--configuration", configuration)
+                    .WorkingDirectory(Path.Combine(Dirs.RepoRoot, "test", testProject))
+                    .Execute()
+                    .EnsureSuccessful();
+            }
+            return c.Success();
+        }
+
+        [Target]
+        public static BuildTargetResult RunTests(BuildTargetContext c)
+        {
+            var dotnet = DotNetCli.Stage0;
+
+            var configuration = c.BuildContext.Get<string>("Configuration");
+
+            CopyTestProjectsToArtifacts();
+            
+            var failingTests = RunDotnetTestOnTestProjects(c, dotnet, configuration);
+            if (failingTests.Any())
+            {
+                foreach (var project in failingTests)
+                {
+                    c.Error($"{project} failed");
+                }
+                return c.Failed("Tests failed!");
+            }
+
+            return c.Success();
+        }
+
+        private static List<string> RunDotnetTestOnTestProjects(BuildTargetContext c, DotNetCli dotnet, string configuration)
+        {
+            var failingTests = new List<string>();
+
+            foreach (var project in TestProjects)
+            {
+                c.Info($"Running tests in: {project}");
+
+                var result = dotnet.Test("--configuration", configuration, "-xml", $"{project}-testResults.xml", "-notrait", "category=failing")
+                    .WorkingDirectory(Path.Combine(Dirs.RepoRoot, "test", project))
+                    .EnvironmentVariable("PATH", $"{dotnet.BinPath}{Path.PathSeparator}{Environment.GetEnvironmentVariable("PATH")}")
+                    .EnvironmentVariable("TEST_ARTIFACTS", Dirs.TestArtifacts)
+                    .Execute();
+
+                if (result.ExitCode != 0)
+                {
+                    failingTests.Add(project);
+                }
+            }
+
+            return failingTests;
+        }
+
+        private static void CopyTestProjectsToArtifacts()
```

Comment:
There is no TestAssetsManager. I did not bring over the CLI test utils, and this class is more simple and covers the set of functionality that we need.

---
In reply to: [66099798](https://github.com/dotnet/core-setup/pull/97#discussion_r66099798) [](ancestors = 66099798)